src/Microsoft.ML.HalLearners/LRWithTrainingStatistics.cs

```diff
@@ -0,0 +1,346 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.ML.Runtime;
+using Microsoft.ML.Runtime.Data;
+using Microsoft.ML.Runtime.EntryPoints;
+using Microsoft.ML.Runtime.Internal.Internallearn;
+using Microsoft.ML.Runtime.Internal.Utilities;
+using Microsoft.ML.Runtime.Learners;
+using Microsoft.ML.Runtime.Numeric;
+using Microsoft.ML.Runtime.Training;
+using Microsoft.ML.Trainers.HalLearners;
+using System;
+using System.Collections.Generic;
+
+[assembly: LoadableClass(LogisticRegressionWithStats.Summary, typeof(LogisticRegressionWithStats), typeof(LogisticRegression.Arguments),
+    new[] { typeof(SignatureBinaryClassifierTrainer), typeof(SignatureTrainer), typeof(SignatureFeatureScorerTrainer) },
+    LogisticRegressionWithStats.UserNameValue,
+    LogisticRegressionWithStats.LoadNameValue,
+    LogisticRegressionWithStats.ShortName,
+    "logisticregressionwrapper")]
+
+[assembly: LoadableClass(typeof(void), typeof(LogisticRegressionWithStats), null, typeof(SignatureEntryPointModule), LogisticRegressionWithStats.LoadNameValue)]
+
+namespace Microsoft.ML.Trainers.HalLearners
+{
+    using Mkl = OlsLinearRegressionTrainer.Mkl;
+
+    /// <include file='doc.xml' path='doc/members/member[@name="LBFGS"]/*' />
+    /// <include file='doc.xml' path='docs/members/example[@name="LogisticRegressionBinaryClassifier"]/*' />
+    public sealed partial class LogisticRegressionWithStats : LogisticRegression
```

Comment:
(nit) In general I think the file name should match the class name. Here we have `LogisticRegressionWithStats` vs `LRWithTrainingStatistics.cs`